addons/hr_work_entry/models/hr_work_entry.py

```diff
@@ -66,6 +67,49 @@ class HrWorkEntry(models.Model):
     def init(self):
         tools.create_index(self._cr, "hr_work_entry_date_start_date_stop_index", self._table, ["date_start", "date_stop"])
 
+    def _get_actual_durations(self, work_entries):
+        return sum(work_entry.duration for work_entry in work_entries)
+
+    def _get_work_entry_calendar(self, value):
+        return value.resource_calendar_id
+
+    def _get_work_entries_data(self, work_entries_dict):
+        data = self.env['hr.work.entry']
+        for key, work_entries in work_entries_dict.items():
+            actual_duration = self._get_actual_durations(work_entries)
+            week_type = "0" if key[0].isocalendar()[1] % 2 == 0 else "1"
+            work_entry_calendar = self._get_work_entry_calendar(key[1])
+            filter_condition = lambda x: (not work_entry_calendar.two_weeks_calendar or (x.week_type == week_type)) and int(x.dayofweek) == key[0].weekday() and x.day_period != 'lunch'
```

Comment:
for readability purposes
```suggestion
            filter_condition = lambda x:\
                (not work_entry_calendar.two_weeks_calendar or (x.week_type == week_type))\
                and int(x.dayofweek) == key[0].weekday()\
                and x.day_period != 'lunch'
```